lightning/src/sync.rs

```diff
@@ -34,11 +33,11 @@ impl<T> Mutex<T> {
 		Mutex { inner: RefCell::new(inner) }
 	}
 
-	pub fn lock(&self) -> LockResult<MutexGuard<'_, T>> {
+	pub fn lock<'a>(&'a self) -> LockResult<MutexGuard<'a, T>> {
```

Comment:
Could you move the explicit lifetimes into the previous commit?

Reply:
squashing into a single commit